### PR TITLE
Do not send kms_arn in glue security configuration if mode is SSES3

### LINKED ISSUE #12879: Cannot use aws_glue_security_configuration with SSE-S3 encryption because of spurious empty KmsKeyArn
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.24
AWS provider v2.57

### Affected Resource(s)

* aws_glue_security_configuration

### Terraform Configuration Files

```hcl
resource "aws_glue_security_configuration" "sec_conf" {
  name = "sec_conf"

  encryption_configuration {
    cloudwatch_encryption {
      cloudwatch_encryption_mode = "DISABLED"
    }

    job_bookmarks_encryption {
      job_bookmarks_encryption_mode = "DISABLED"
    }

    s3_encryption {
      s3_encryption_mode = "SSE-S3"
    }
  }
}
```

### Expected Behavior

When I read the resource back with boto3 client, I expect to see the following:

```python
import boto3
client = boto3.client("glue")
client.get_security_configuration(Name="sec_conf")
```

```json
{
  "Name": "sec_conf",
  "EncryptionConfiguration": {
    "S3Encryption": [
      {
        "S3EncryptionMode": "SSE-S3"
      }
    ],
    "CloudWatchEncryption": {
      "CloudWatchEncryptionMode": "DISABLED"
    },
    "JobBookmarksEncryption": {
      "JobBookmarksEncryptionMode": "DISABLED"
    }
  }
}
```

and to be able to use the resource.

### Actual Behavior

It is created with empty strings for "KmsKeyArn"

```json
{
  "Name": "sec_conf",
  "EncryptionConfiguration": {
    "S3Encryption": [
      {
        "S3EncryptionMode": "SSE-S3",
        "KmsKeyArn": ""
      }
    ],
    "CloudWatchEncryption": {
      "CloudWatchEncryptionMode": "DISABLED",
      "KmsKeyArn": ""
    },
    "JobBookmarksEncryption": {
      "JobBookmarksEncryptionMode": "DISABLED",
      "KmsKeyArn": ""
    }
  }
}
```

Trying to use the configuration in a glue job, I get the following error message:

```
An error occurred while calling z:com.amazonaws.services.glue.util.Job.commit. 1 validation error detected: Value '' at 'keyId' failed to satisfy constraint: Member must have length greater than or equal to 1 (Service: AWSKMS; Status Code: 400; Error Code: ValidationException; Request ID: ...)
```

### Workaround attempts

1. I tried setting the `kms_key_arn = null` in the terraform, but the empty string is still there.

2. I also tried to specify the key arn, as follows:

```hcl

data "aws_kms_alias" "s3" {
  name = "alias/aws/s3"
}

resource "aws_glue_security_configuration" "sec_conf" {
  name = "sec_conf"

  encryption_configuration {
    cloudwatch_encryption {
      cloudwatch_encryption_mode = "DISABLED"
    }

    job_bookmarks_encryption {
      job_bookmarks_encryption_mode = "DISABLED"
    }

    s3_encryption {
      s3_encryption_mode = "SSE-S3"
      kms_key_arn = data.aws_kms_alias.s3.target_key_arn
    }
  }
}
```

In that case, `terraform apply` gives me the following error message:

```
Error: error creating Glue Security Configuration (glue_security): InvalidInputException: kmsKeyArn can only be empty for s3EncryptionMode that is not SSE_KMS

  on main.tf line 6, in resource "aws_glue_security_configuration" "sec_conf":
   6: resource "aws_glue_security_configuration" "sec_conf" {

```

### Steps to Reproduce

1. `terraform apply`
2. Use the configuration in a job

### PR DESCRIPTION
**Pull request by maparent**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #12879

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSES3'
```

### CODE REVIEW

File: aws/resource_aws_glue_security_configuration.go [hunk 1]
```diff
@@ -272,8 +272,10 @@ func expandGlueS3Encryption(m map[string]interface{}) *glue.S3Encryption {
 		S3EncryptionMode: aws.String(m["s3_encryption_mode"].(string)),
 	}
 
-	if v, ok := m["kms_key_arn"]; ok {
```
Comment (bflad) [inline aws/resource_aws_glue_security_configuration.go:275]: Hi @maparent 👋 Does it work if we instead do a `v.(string) != ""` check here? Introducing the additional `*s3Encryption.S3EncryptionMode != glue.S3EncryptionModeSseS3` conditional can lead to confusing perpetual differences (and if the configuration of both with real values is invalid, the API error can be returned)

```go
if v, ok := m["kms_key_arn"]; ok && v.(string) != "" {
```

Comment (maparent) [reply aws/resource_aws_glue_security_configuration.go:275]: Yes sounds like a great solution, sorry I missed it.

Comment (maparent) [reply aws/resource_aws_glue_security_configuration.go:275]: Not at my computer, will alter the pr in 1h if you can’t.

**Comment by maparent** [comment]
(sent the modification)

**Pull request merged**